File: core/src/main/java/com/pluralsight/hydra/expr/Expressions.java
package com.pluralsight.hydra.expr;

import org.joda.time.DateTime;

/**
 * Created by alexsilva on 7/7/17.
 */
public class Expressions {

    public static String current_timestamp() {
        return String.valueOf(System.currentTimeMillis());
    }

    public static int this_month() {
        return DateTime.now().getMonthOfYear();
    }
}

File: core/src/main/scala/hydra/spark/util/Expressions.scala
package hydra.spark.util

import org.springframework.expression.common.TemplateParserContext
import org.springframework.expression.spel.standard.SpelExpressionParser
import org.springframework.expression.spel.support.StandardEvaluationContext

/**
  * Created by alexsilva on 7/7/17.
  */
object Expressions {
  private val evalCtx = new StandardEvaluationContext()

  private val exprs = new com.pluralsight.hydra.expr.Expressions()

  def parseExpression(expr: String): String = {
    val parser = new SpelExpressionParser()
    parser.parseExpression(expr, new TemplateParserContext()).getValue(evalCtx, exprs, classOf[String])
  }

}

